File: 02-Backend/spring-boot-ecommerce/spring-boot-ecommerce/src/main/java/com/watches/ecommerce/dto/Purchase.java
package com.watches.ecommerce.dto;

import com.watches.ecommerce.entity.Address;
import com.watches.ecommerce.entity.Customer;
import com.watches.ecommerce.entity.Order;
import com.watches.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;

    private Address shippingAddress;

    private Address billingAddress;

    private Order order;

    private Set<OrderItem> orderItems;





}

File: 01-Frontend/angular-ecommerce/src/app/common/purchase.ts
import { OrderItem } from './order-item';
import { Customer } from "./customer";
import { Address } from './address';
import { Order } from './order';

export class Purchase {

    customer!: Customer; 
    shippingAddress!: Address 
    billingAddress!: Address; 
    order!: Order; 
    orderItems!: OrderItem[];
 



}
